### PR TITLE
Update chat extension guide with GitHub App integration details

### PR DESCRIPTION
**Pull request by isidorn**

### CODE REVIEW

File: api/extension-guides/chat.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ MetaDescription: A guide to creating an AI extension in Visual Studio Code
 
 Visual Studio Code's Copilot Chat architecture enables extension authors to integrate with the [Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) experience. A chat extension is a VS Code extension that uses the Chat extension API by contributing a *Chat participant*.
 
+It is also possible to extend GitHub Copilot by contributing a [GitHub App](https://github.com/marketplace). A GitHub App is backed by a service, and will work across all the GitHub Copilot surfaces (github.com, Visual Studio, VS Code). GitHub Apps do not have full access to the VS Code API. To extend GitHub Copilot through a GitHub App you should join the [Copilot Partner Program](https://github.com/features/preview/copilot-partner-program).
```
Comment (ntrogh) [inline api/extension-guides/chat.md:14]:
```suggestion
It is also possible to extend GitHub Copilot by contributing a [GitHub App](https://github.com/marketplace). A GitHub App is backed by a service and works across all GitHub Copilot surfaces, such as github.com, Visual Studio, or VS Code. GitHub Apps do not have full access to the VS Code API. To extend GitHub Copilot through a GitHub App, you should join the [Copilot Partner Program](https://github.com/features/preview/copilot-partner-program).
```

**Review by DonJayamanne — APPROVED** [review]

**Review by ntrogh — CHANGES_REQUESTED** [review]
Minor text updates.

**Comment by isidorn** [comment]
I have applied the changes, and this is ready for another review. Thank you Nick!

**Review by ntrogh — APPROVED** [review]
Good stuff!

**Pull request merged**